fili-core/src/main/java/com/yahoo/bard/webservice/logging/TimedPhase.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a phase that is timed.
+ * TimedPhase is used to associate a Timer located in the registry with the exact duration of such a phase for a
+ * specific request.
+ */
+public class TimedPhase implements AutoCloseable {
+    private static final Logger LOG = LoggerFactory.getLogger(TimedPhase.class);
+
+    private final String name;
+    private long start;
+    private long duration;
```

Comment:
I'm ok with making it thread safe, especially since many of these timers are stopped on a different thread than they're started (the request vs response thread). Rethinking the semantics is outside the scope of this PR though, I think.